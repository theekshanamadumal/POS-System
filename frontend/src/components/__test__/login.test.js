import React from 'react';
import Login from '../login/login';
import { render, screen,fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

 
describe("<Login />", () => {
  
  test('render email input', () => {
    const {queryByTestId}=render(<Login />);
    const lg=queryByTestId("login");
    expect(lg).toBeTruthy();
  });

  test("heading detail",()=>{
    const {queryByTestId}=render(<Login />);
    const lg=queryByTestId("heading");
    expect(lg.textContent).toEqual("Welcome To POS System!");
  })
 
  test('pass valid email to test email input field', () => {
    render(<Login />);
    const inputEl = screen.getByTestId("email");
    userEvent.type(inputEl, "test@mail.com");
    expect(screen.queryByTestId("emailStatus").textContent).toEqual("");
  });
 
  test('pass invalid email to test input value', () => {
    const {queryByTestId}=render(<Login/>);
    const inputEl = screen.queryByTestId("email");
    fireEvent.change(inputEl,{target:{value:"sara"}});
    userEvent.type(inputEl, "");
    expect(screen.queryByTestId("emailStatus").textContent).toEqual("");
  });
  test("snapshot checking",()=>{
    const {container}=render(<Login />);
    expect(container.firstChild).toMatchSnapshot();
  })
  const onSubmit=jest.fn()
 
  test("onSubmit function",()=>{
    const {queryByTestId,getByLabelText}=render(<Login onSubmit={onSubmit} />);
    fireEvent.click(queryByTestId("submitForm"));
    //getByLabelText("Email address").value="hello@gmail";
    //fireEvent.change(getByLabelText("Email address"))
    //console.log(getByLabelText("Email address").outerHTML)
    //expect(queryByTestId("submitForm")).toHaveBeenCalledTimes(1);
  })
 
});
