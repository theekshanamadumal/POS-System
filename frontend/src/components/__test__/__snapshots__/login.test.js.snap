// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> snapshot checking 1`] = `
<div
  class="bgImg"
  data-testid="login"
>
  <h1
    class="text-center p-4 mt-4 text-light"
    data-testid="heading"
  >
    Welcome To POS System!
  </h1>
  <div
    class="d-flex justify-content-center"
  >
    <div
      class="mx-2 loginBox p-4"
    >
      <form>
        <h3
          class="text-center"
        >
          Sign In
        </h3>
        <img
          class="align-center userLogo my-2"
          src="/images/userlogo.png"
        />
        <div
          class="form-group form-out"
        >
          <label
            for="id"
          >
            Email address
          </label>
          <input
            class="form-control"
            data-testid="email"
            id="email"
            name="Email"
            placeholder="Enter email"
            value=""
          />
          <small
            data-testid="emailStatus"
          />
        </div>
        <div
          class="form-out form-group"
        >
          <label>
            Password
          </label>
          <input
            class="form-control pass"
            data-testid="password"
            id="password"
            name="Password"
            placeholder="Enter password"
            type="password"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="position: absolute; margin-top: -30px; margin-left: 230px; color: rgb(97, 96, 96);"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
          <small />
        </div>
        <div
          class="form-group"
        >
          <div
            class="custom-control custom-checkbox"
          >
            <input
              class="custom-control-input"
              id="customCheck1"
              type="checkbox"
            />
          </div>
        </div>
        <button
          class="btn btn-block text-light lgBtn"
          data-testid="loginBtn"
          style="background-color: rgb(46, 100, 100); margin-top: -20px;"
          type="submit"
        >
          Sign In
        </button>
        <p
          class="forgot-password text-right"
        >
          Forgot 
          <a
            href="#"
          >
            password?
          </a>
        </p>
      </form>
    </div>
  </div>
</div>
`;
